contrib/googlejavaformat/tests/python/pants_test/contrib/googlejavaformat/test_googlejavaformat.py

```diff
@@ -0,0 +1,49 @@
+# coding=utf-8
+# Copyright 2018 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import (absolute_import, division, generators, nested_scopes, print_function,
+                        unicode_literals, with_statement)
+
+from pants_test.pants_run_integration_test import PantsRunIntegrationTest
+
+
+TEST_DIR = 'contrib/googlejavaformat/testprojects/src/java/org/pantsbuild/testproject'
+
+
+class GoogleJavaFormatIntegrationTests(PantsRunIntegrationTest):
```

Comment:
Since ScalaRewriteBase tasks are self contained and require no product setup it would be a good deal nicer to model this as a `pants_test.tasks.task_test_base.TaskTestBase` unit test instead where `make_target` and `create_file` were used to emit ephemeral source code to lint and re-format. Faster tests plus no muddying checked in files. What do you think? Could be a follow up conversion too.

Reply:
Sounds good - I was just cribing from `test_scalafmt.py` :) Will make this change.

Reply:
Fixed that one!: https://github.com/pantsbuild/pants/pull/5599
Down from ~33s wall time in py.test to ~10s plus now cacheable.